configuration/src/main/java/com/synopsys/integration/configuration/config/PropertyMasker.java

```diff
@@ -0,0 +1,52 @@
+package com.synopsys.integration.configuration.config;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.synopsys.integration.configuration.property.Property;
+
+public class PropertyMasker {
+    // TODO - do we need this?
+    public Map<Property, String> maskRawValues(Set<Property> knownProperties, Map<String, String> rawKeyValues, Predicate<String> shouldMask) {
```

Comment:
I would be fine with having getMaskedRaw on property config that returns `Map<Property, String>` instead of this class.  if that is what the TODO means. I don't know if I am the reason this is a separate class or not :). 

I'd say it seems like we are doing an awfully large amount of work here.

Reply:
This method is unused in my changes, I included it just as a possible utility, but it should probably just be deleted.